roles/deploy_bmh/template/bmh.yml.j2

```diff
@@ -0,0 +1,41 @@
+{%- for node in baremetal_nodes if node.split('-')[0] not in [ 'crc', 'controller', 'ocp' ]  %}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ node }}-bmc-secret
+  namespace: openstack
+type: Opaque
+stringData:
+  username: {{ baremetal_nodes[node]['username'] }}
+  password: {{ baremetal_nodes[node]['password'] }}
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: {{ node }}
+  namespace: openstack
```

Comment:
this namespace is not watched by metal3. Any BMH created outside it seems to be ignored.
We should allow to parameterize this one (and the associated secret namespace)

Reply:
FTR: BMO watches the `ns` set here https://github.com/metal3-io/baremetal-operator/blob/main/main.go#L135, that, in dev-scripts deployments at least, is just this:
```yaml
- name: WATCH_NAMESPACE
  valueFrom:
    fieldRef:
      apiVersion: v1
      fieldPath: metadata.namespace
```

Meaning: BMO watches for resources only inside it's own namespace. All the BMH we will create for VAs need to be applied in the `openshift-machine-api` ns.

Reply:
That is what we encountered when using Metal3 with OSP Director operator as well.  It seemed that we had to put our `Secret`s and `BaremetalHost`s in the `openshift-machine-api` namespace.

@rabi Did you find a way to get the BMO to watch other namespaces?  I thought I recalled you figuring something out that allowed `openstack` namespace to be used.

Reply:
Yes,oc patch provisioning provisioning-configuration --type merge -p '{"spec":{"watchAllNamespaces": true }}'`

Reply:
Thank you both for that feedback and pointer.